### PR TITLE
use AWS ES with AWS request signing

### PR DESCRIPTION
**Pull request by nusnewob**

### CODE REVIEW

File: config/settings/common.py [hunk 1]
```diff
@@ -184,7 +184,11 @@
 APPEND_SLASH = False
 
 # Leeloo stuff
+AWS_ELASTICSEARCH_REGION = env('AWS_ELASTICSEARCH_REGION')
```
Comment (reupen) [inline config/settings/common.py]: Don't we need a ES_USE_AWS_AUTH setting as we won't use AWS auth locally etc. and we'll need to be able to manage the switchover as well?

File: datahub/search/elasticsearch.py [hunk 2]
```diff
@@ -80,10 +83,28 @@ class MatchNone(Query):
...
+        aws_secret_access_key = settings.AWS_ELASTICSEARCH_SECRET,
+        aws_host = es_host.netloc,
+        aws_region = settings.AWS_ELASTICSEARCH_REGION,
+        aws_service = 'es'
+    )
     connections.configure(
         default={
-            'hosts': [settings.ES_URL],
-            'verify_certs': settings.ES_VERIFY_CERTS
+            'hosts': [es_host.netloc],
```
Comment (reupen) [inline datahub/search/elasticsearch.py]: Doesn't passing the URL work with AWS auth?

File: datahub/search/elasticsearch.py [hunk 3]
```diff
@@ -80,10 +83,28 @@ class MatchNone(Query):
...
+            'port': es_port,
+            'use_ssl': settings.ES_USE_SSL,
```
Comment (reupen) [inline datahub/search/elasticsearch.py]: Can't we get this from http vs https in the URL?

File: datahub/search/elasticsearch.py (hunk 2 above)
Comment (nusnewob) [reply datahub/search/elasticsearch.py]: AWS auth requires the signing URL and request URL to be exact match (without http/https and port)

File: datahub/search/elasticsearch.py (hunk 3 above)
Comment (nusnewob) [reply datahub/search/elasticsearch.py]: `urlparse` doesn't return port if only http/https specified.

File: datahub/search/elasticsearch.py (hunk 2 above)
Comment (nusnewob) [reply datahub/search/elasticsearch.py]: `ES_USE_AWS_AUTH` switch added in new commit.

File: config/settings/common.py (hunk 1 above)
Comment (nusnewob) [reply config/settings/common.py]: `ES_USE_AWS_AUTH` switch added in new commit.

Comment (reupen) [reply config/settings/common.py]: So can we only load AWS_ELASTICSEARCH_REGION/AWS_ELASTICSEARCH_KEY/AWS_ELASTICSEARCH_SECRET if ES_USE_AWS_AUTH is True?

File: datahub/search/elasticsearch.py (hunk 3 above)
Comment (reupen) [reply datahub/search/elasticsearch.py]: Yes but can't we determine USE_SSL by checking if the URL scheme is https?

File: config/settings/common.py (hunk 1 above)
Comment (nusnewob) [reply config/settings/common.py]: yep, updated it

File: datahub/search/elasticsearch.py (hunk 3 above)
Comment (nusnewob) [reply datahub/search/elasticsearch.py]: not set by default now, only loaded when `ES_USE_AWS_AUTH` is True, it's default protocol on AWS ES.

**Comment by reupen** [comment]
@nusnewob There's still a failing test.

File: datahub/search/elasticsearch.py [hunk 4]
```diff
@@ -1,7 +1,11 @@
 from collections import defaultdict
 from itertools import chain
 
+from urllib.parse import urlparse
+from aws_requests_auth.aws_auth import AWSRequestsAuth
```
Comment (reupen) [inline datahub/search/elasticsearch.py]: The urllib import should be in the first group (standard library imports) and the aws one in the next group (third-party imports). So it should look like:

```
from collections import defaultdict
from itertools import chain
from urllib.parse import urlparse

from aws_requests_auth.aws_auth import AWSRequestsAuth
from django.conf import settings
from elasticsearch import RequestsHttpConnection
...
```

**Pull request merged**